src/test/kotlin/com/ampnet/identityservice/service/FaucetQueueServiceTest.kt

```diff
@@ -75,7 +110,89 @@ class FaucetQueueServiceTest : TestBase() {
         }
 
         verify("Service will handle the task") {
-            waitUntilTasksAreProcessed()
+            processAllTasks()
+
+            val tasks = faucetTaskRepository.findAll()
+
+            assertThat(tasks).hasSize(1)
+            assertThat(tasks.first().status).isEqualTo(FaucetTaskStatus.COMPLETED)
+            assertThat(tasks.first().hash).isEqualTo(hash)
+        }
+    }
+
+    @Test
+    fun mustRetryPendingTaskWhenExceptionIsThrown() {
+        suppose("Blockchain service will throw exception") {
+            given(blockchainService.sendFaucetFunds(any(), any())).willThrow(RuntimeException())
+        }
+
+        suppose("There is a task in queue") {
+            createFaucetTask(status = FaucetTaskStatus.CREATED)
+        }
+
+        verify("Task will fail and new task will be created") {
+            processAllTasks(maxAttempts = 1)
+
+            val tasks = faucetTaskRepository.findAll()
+
+            assertThat(tasks).hasSize(2)
+            assertThat(tasks.first().status).isEqualTo(FaucetTaskStatus.FAILED)
+            assertThat(tasks[1].status).isEqualTo(FaucetTaskStatus.CREATED)
```

Comment:
Maybe remove `first` because it is tricky to read with number `1` below:
```suggestion
            assertThat(tasks.[0].status).isEqualTo(FaucetTaskStatus.FAILED)
            assertThat(tasks[1].status).isEqualTo(FaucetTaskStatus.CREATED)
```